src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2556,6 +2556,37 @@ theorem not_extractLsb'_append_not_extractLsb'_eq_not_extractLsb' {x : BitVec w}
     congr 1
     omega
 
+/-- A sign extension of `x : BitVec w`
+equals high bits of either `0` or `1` depending on `x.msb`,
+followed by the low bits of `x`. -/
+theorem signExtend_eq_append_extractLsb' {w v : Nat} {x : BitVec w} :
+    x.signExtend v =
+    ((if x.msb then allOnes (v - w) else 0#(v - w)) ++ x.extractLsb' 0 (min w v)).cast (by omega) := by
+  ext i hi
+  simp only [getElem_cast]
+  by_cases hx : x.msb
+  · simp only [signExtend_eq_not_setWidth_not_of_msb_true hx, getElem_not, getElem_setWidth,
+    getLsbD_not, Bool.not_and, Bool.not_not, hx, ↓reduceIte, getElem_append, getElem_extractLsb',
+    Nat.zero_add, getElem_allOnes, dite_eq_ite, Bool.if_true_right]
```

Comment:
```suggestion
      getLsbD_not, Bool.not_and, Bool.not_not, hx, ↓reduceIte, getElem_append, getElem_extractLsb',
      Nat.zero_add, getElem_allOnes, dite_eq_ite, Bool.if_true_right]
```
Formatting nit: please indent multi-line simps, so that it's easier to see where the next tactic starts (also at other simps in the PR!)